forge-gui/src/main/java/forge/gamemodes/match/input/InputConfirmMulligan.java

```diff
@@ -117,7 +117,7 @@ protected boolean onCardSelected(final Card c0, final List<Card> otherCardsToSel
             cardSelectLocked = true;
             ThreadUtil.invokeInGameThread(new Runnable() {
                 @Override public void run() {
-                    final CardCollectionView hand = c0.getController().getCardsIn(ZoneType.Hand);
+                    final CardCollection hand = new CardCollection(c0.getController().getCardsIn(ZoneType.Hand));
```

Comment:
this should work rather than instantiating new objects
```
                    final CardCollectionView hand = c0.getController().getCardsIn(ZoneType.Hand);
                    final int size = hand.size();
                    for (final Card c : hand.threadSafeIterable()) {
                        player.getGame().getAction().exile(c, null);
                    }
                    c0.getController().drawCards(size);
                    cardSelectLocked = false;
```

Reply:
hmm, so whats the best method to fix this @Hanmac

Reply:
imo threadSafeIterable might be better than full CardCollection

Reply:
yea that just makes it clearer to the reader what the intent is

Reply:
ty; implemented this suggestion.   I was looking to simplify for readability, and was wondering if we even need the "fromHand" check?  At this step, it'll always be from hand.
  1) Game starts  2) Serum Powder checks  3) mulligans 4) possible jump back to second step  5) Hand secured 6) interactions with Leylines/Gemstone Caverns/Chancellors/Providence/Sphinx of Foresight